python/mpi_cmake_modules/python_format.py

```diff
@@ -0,0 +1,109 @@
+""" python_format
+
+Utility functions for creating formatting script based on clang 
+
+License BSD-3-Clause
+Copyright (c) 2019, New York University and Max Planck Gesellschaft.
+"""
+
+import argparse
+import sys
+import os
+from os import walk
+from os import path
+
+try:
+    from shutil import which
+except:
+    import distutils.spawn
+
+    ## This ensure the compatibility Python2 vs Python3
+    which = distutils.spawn.find_executable
+
+from mpi_cmake_modules.utils import parse_args
+
+
+def _find_black():
+    """Find the full path to the clang-format executable.
+
+    Look by default for the `black` executable in the PATH environment
+    variable.
+
+    Returns:
+        The full path to the black executable.
+    """
+    path_to_black = which("black")
+    if path_to_black is not None:
+        return path_to_black
+    raise Exception(
+        "black executable not found. You may try "
+        "'(sudo -H) pip install install black'"
+    )
+
+
+def _execute_black(
+    black_format_bin, list_of_files=None, list_of_directories=None
+):
+    """Execute the formatting of python files using black.
+
+    Get the path to the executable, and run it on the list of files to format.
+
+    Args:
+        black_format_bin (str):  Path to the black binary.
+
+        black_format_config (list(str)): One line dictionnary string containing
+        the black parameters.
+
+        black_format_arg list(str): List of source files to parse.
+    """
+    arg_files = ""
+    if list_of_files is not None:
+        arg_files = " --pyi " + " ".join(list_of_files)
```

Comment:
I didn't really understand the documentation of the `--pyi` argument but after a quick test, I think we should not use it.  It removes a lot of empty lines which should be kept to be pep8-compliant (e.g. between class methods).

Reply:
Without this, I assume there is also no need anymore to separate files and directories in the arguments.

Reply:
The problem is how do you format scripts?
the "--pyi" is just saying please include these files to the formatting.

Reply:
My bad you are right